storage/rocksdb/CMakeLists.txt

```diff
@@ -266,7 +266,7 @@ IF (ROCKSDB_DYNAMIC_PLUGIN)
   MYSQL_ADD_PLUGIN(rocksdb ${ROCKSDB_SOURCES} STORAGE_ENGINE DEFAULT
     MODULE_ONLY LINK_LIBRARIES ${rocksdb_static_libs})
 ELSE()
-  MYSQL_ADD_PLUGIN(rocksdb_se ${ROCKSDB_SOURCES} STORAGE_ENGINE DEFAULT
+  MYSQL_ADD_PLUGIN(rocksdb_se ${ROCKSDB_SOURCES} STORAGE_ENGINE MANDATORY
```

Comment:
The ability to build RocksDB as a dynamic plugin remains - which is incompatible with native DD and is likely to be broken by the `sql_plugin.cc` changes. As a minimum, please test this PR with `-DWITH_ROCKSDB_DYNAMIC_PLUGIN=ON` (not sure if that's the exact CMake option name), and then please either make any changes to make it work, either remove the dynamic plugin option. But the latter may break some user scripts for a feature that does not exist yet.

Reply:
In the diff tool, we have test set for both dynamicRocksdb and static rocksdb. From the test result, most of them passed. There are few test case failure. But I don't think it's related to the change. @luqun Could you please help to verify it's not related to this diff's change?

Reply:
I will take a look this part to see whether it works with rocksdb dynamic plugin..